helpers/junos_helpers/serial.go

```diff
@@ -386,21 +340,8 @@ func (g *GoNCClient) ReadRawGroup(applygroup string) (string, error) {
 	return reply.Data, nil
 }
 
-func publicKeyFile(file string) ssh.AuthMethod {
-	buffer, err := ioutil.ReadFile(file)
-	if err != nil {
-		return nil
-	}
-
-	key, err := ssh.ParsePrivateKey(buffer)
-	if err != nil {
-		return nil
-	}
-	return ssh.PublicKeys(key)
-}
-
-// NewClient returns gonetconf new client driver
-func NewClient(username string, password string, sshkey string, address string, port int) (*GoNCClient, error) {
+// NewSerialClient returns gonetconf new client driver
+func NewSerialClient(username string, password string, sshkey string, address string, port int) (NCClient, error) {
```

Comment:
By NewSerialClient, what do you mean exactly? Serial as in sequential?

Reply:
perhaps the naming isn't great, but since the current flow is using mutex locking and unlocking it ends up so that only one action is performed at the same time so I went with the name serial.